src/material/datepicker/date-range-input-parts.ts

```diff
@@ -181,6 +188,26 @@ abstract class MatDateRangeInputPartBase<D>
     ) as MatDateRangeInputPartBase<D> | undefined;
     opposite?._validatorOnChange();
   }
+
+  protected _onKeyup(event: KeyboardEvent) {
+    this._onKeyupSubject.next(event);
+  }
+
+  protected _moveFocusOnKeyup(
```

Comment:
Why are we using `keyup` here instead of the existing `keydown` listener? We already have a similar navigation for when the user presses backspace on an empty end input.

Reply:
If you only listen on keydown then move the focus, the keyup event will fire against the newly focused field and move the cursor. I wait until after keyup fires to move focus so that it will not fire against the other field.

Simply listening on keyup does not work well either as you'll get the event after the cursor has moved thus cannot tell if it was originally at the start/end of the input or whether it just arrived there as part of that event.

Reply:
Maybe I'm missing something, but does it matter that the `keyup` event fires against the newly-focused input field? If we moved all the logic into `keydown`, we wouldn't have a `keyup` listener at all.

I tried copying over your code and only using `keydown` events and it was working as expected. The one additional call I had to add was `preventDefault` so that the caret doesn't move immediately to the next character after focus has been moved. 

Here's my change: https://github.com/angular/components/commit/06f1cc55e4278af457236dda750ab40ca22b183f

Reply:
I'll give that a try

Reply:
Ok, preventDefault did the trick. Removed all the keyup stuff.